bob/run.go

```diff
@@ -145,6 +150,37 @@ func buildNonInteractive(ctx context.Context, runname string, aggregate *bobfile
 		return ErrRunDoesNotExist
 	}
 
+	var tasksInPipeline []string
+	nixDependencies := make([]nix.Dependency, 0)
+	for _, child := range interactive.DependsOn {
+		if isInteractive(child, aggregate) {
+			continue
+		}
+		err = aggregate.BTasks.CollectTasksInPipeline(child, &tasksInPipeline)
+		if err != nil {
+			return err
+		}
+		err = aggregate.BTasks.CollectNixDependencies(child, &nixDependencies)
+		if err != nil {
+			return err
+		}
+	}
+
+	if len(nixDependencies) > 0 {
+		fmt.Println("Building nix dependencies...")
+
+		storePaths, err := BuildNixDependencies(nix.UniqueDeps(append(nix.DefaultPackages(), nixDependencies...)))
+		if err != nil {
+			return err
+		}
+
+		for _, name := range tasksInPipeline {
+			t := aggregate.BTasks[name]
+			t.SetStorePaths(sliceutil.Unique(storePaths))
```

Comment:
Sure this is correct? Doesn't this set all storePaths.. even the ones defined on another task?

Reply:
Seems to me that got lost in this commit https://github.com/benchkram/bob/pull/46/commits/fe55739f87f13c4191b7143740f76eb5886b7946#diff-b3ed69358a144efc695faf256e282c7e89a1043afbaac51b6ee0ec0a88ebbcb9L62

Reply:
I changed this following this conversation https://benchkram.slack.com/archives/D039GK6PRFH/p1651822002490399

because I will need store paths from all tasks in pipeline, not only the ones set in task dependencies and bobfile dependencies.